lib/src/services/bedrock_service.dart

```diff
@@ -47,9 +84,26 @@ class BedrockService {
     dio.options.receiveTimeout = 100000;
   }
 
-  Future<Response> post(String endpoint, {Map payload = const {}}) async {
-    return await dio.post(endpoint, data: payload);
+  Future<dynamic> post(String url,
+      {dynamic body, Map<String, dynamic>? queryParams}) async {
+    var response =
+        await dio.post(url, data: body, queryParameters: queryParams);
+    return response.data;
```

Comment:
I'm not sure if you were able to test this locally but it currently breaks due to returning the `.data` rather than the full response. The `DioLogger` also throws an error because it tries to print some attributes coming from the `response.data` but the returned value also has data unwrapped. Is there a particular reason for returning data rather than the full response back?